docs/fastdds/xtypes/language_binding.rst

```diff
@@ -0,0 +1,1000 @@
+facilitate the creation of this type.
+This API requires the type stored in the collection and the sequence with the collection's dimensions.
+
+|DynamicData-api| class provides specific :func:`get_values` and :func:`set_values` functions for each primitive
+type, allowing users to easily work with arrays of primitives types.
+For arrays of more complex types, please refer to :ref:`xtypes_complextypes`.
+
+.. note::
+
+    Multi-dimensional arrays *flatten* every dimension into a single-dimension array.
+
+Primitive type promotion is also applicable for these methods.
+
+If a specific range of values within the array are to be modified, passing the starting index to
+:func:`get_values` / :func:`set_values` would only manage data from that element forward until the length of the given
+input.
+Specific collection's element can be also be modified using the :func:`get_value` / :func:`set_value` passing the index
+of the element to be modified.
+
+.. tabs::
+
+    .. tab:: IDL
+
+        .. literalinclude:: /../code/DynamicTypesIDLExamples.idl
+            :language: idl
+            :start-after: //!--IDL_ARRAYS
+            :end-before: //!--
+
+    .. tab:: XML
+
+        .. literalinclude:: /../code/XMLTester.xml
+            :language: xml
+            :start-after: <!-->XML_ARRAYS<-->
+            :end-before: <!--><-->
+
+    .. tab:: C++
+
+        .. literalinclude:: /../code/DDSCodeTester.cpp
+            :language: c++
+            :start-after: //!--CPP_ARRAYS
+            :end-before: //!--
+
+For a detailed explanation about the XML definition of this type, please refer to
+:ref:`XML Array Types<xmldynamictypes_array>`.
+
+.. _xtypes_supportedtypes_map:
+
+Map Types
+^^^^^^^^^
+
+Map types are a collection of key/value pair types.
+Access to the value element is done through the key which is unique within the map type.
+The :code:`TypeKind` used to identify maps is :code:`TK_MAP`.
+|TypeDescriptor-api| :code:`element_type` property must be set with the map value type.
+|TypeDescriptor-api| :code:`key_type` property must be set with the map key type.
+Allowed key types are signed and unsigned integer types and string types.
+
+.. note::
+
+  Currently, wide string keys are not supported as map keys.
+
+Additionally, :code:`bound` property might be configured to set a maximum length for the collection.
+
+|DynamicTypeBuilderFactory-api| exposes the |DynamicTypeBuilderFactory::create_map_type| function to
+facilitate the creation of this type.
+This API requires the type of both the key and the value stored in the collection, and the collection's bound, using
+:code:`LENGTH_UNLIMITED` in case of unbounded maps.
+
+Manipulating map types data is more complex.
+First the :code:`MemberId` corresponding to a specific :code:`key` must be retrieved using
+|DynamicData::get_member_id_by_name| API.
+In order to call this method, the correct :code:`string` representation of the key value must be passed.
```

Comment:
Explain that calling `get_member_id_by_name`, if the key doesn't exists, it will be created. As in `std::map` with operator `[]`.